From patchwork Mon Apr 15 22:14:21 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Nadav Amit <nadav.amit@gmail.com>
X-Patchwork-Id: 10901647
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 1F04D17E6
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon, 15 Apr 2019 22:14:30 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 05A8328864
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon, 15 Apr 2019 22:14:30 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id EDC1528871; Mon, 15 Apr 2019 22:14:29 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 7E3B328864
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon, 15 Apr 2019 22:14:29 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727552AbfDOWO2 (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Mon, 15 Apr 2019 18:14:28 -0400
Received: from mail-wr1-f68.google.com ([209.85.221.68]:42042 "EHLO
        mail-wr1-f68.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726751AbfDOWO2 (ORCPT <rfc822;kvm@vger.kernel.org>);
        Mon, 15 Apr 2019 18:14:28 -0400
Received: by mail-wr1-f68.google.com with SMTP id g3so23965926wrx.9
        for <kvm@vger.kernel.org>; Mon, 15 Apr 2019 15:14:26 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=from:to:cc:subject:date:message-id:mime-version
         :content-transfer-encoding;
        bh=XjO+Y5KRyfX0tjkU3TwennLudgJIbm9Dj/F3EWaypaE=;
        b=DUxIfvVOPr2uqwz12bN9uFSlnKveT0t4mSh1Q6gbsDFGiRS3Zh53+kLfF1rteCTEC+
         N1TEO/Q1uSQDqnndtiiuptWTcu7eypDHmBs7LDx+F10kZMZYEIKUnmLhbIWYoEn39cEJ
         TXk1OrsuI8yp7HYIHDHN2jCzYDIjL0bNnI1tA8/Co9GM3QVN97eS5qeOLWeovF0FwfFM
         oK4jpfLktD4jpcIymQ/SkPKDcSPuXRQ2nlldsDx/loaB+H9meRR95ZWXblmXiU/N/XwQ
         MVuU8gRW4mBOKSzVr9g34YaPd9hmcPtFl+092geqyai3+rQGVEawEy0afHL46PTkkCUG
         z2bg==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:mime-version
         :content-transfer-encoding;
        bh=XjO+Y5KRyfX0tjkU3TwennLudgJIbm9Dj/F3EWaypaE=;
        b=SOudrlzx+4M9ILfdrbgO2s+IjxbURc0DnruzudcFOhh6jIdlyNuJUX43Ijb29vPbB4
         Rj1+LxIoGGelknlAIGlXEI2O+/nz50aFOyCVKFHOFXr+7apPdclmeZ03C7vq1VBvGBMj
         ZvDtA9jH4Rf1YuDyy3/zfo96a+JADTFNiOkVGPIAFrWq6SaEYCytQqN9M88dpdnIU85K
         A41ITyX5XGu5kz4zb9hGPM4OaJRGYsoEL2sdPSRqB1FaZ/TPlT9mbdTJvfj8AbSb175L
         Gq5U38p5zTyY1quZU9Z6p+7eJDCoSIQrbfMvoVIm0wee7voRa06KcaaLMMfRBppxZzVm
         +MKQ==
X-Gm-Message-State: APjAAAXx5G0edpHrLs7o43JvZ7Jokc9xram2JOYtd3ho0hihgCwwAszx
        TZKfVkZ732lJxUer6olK1uU=
X-Google-Smtp-Source: 
 APXvYqw2+f1pjGHpCvlZEKTYh/LviurDXZqG6RUINcFML4k9sEqp5QM8s+9MbEGK8MxSKoQnRdlGYg==
X-Received: by 2002:adf:ce8c:: with SMTP id r12mr29687354wrn.60.1555366465840;
        Mon, 15 Apr 2019 15:14:25 -0700 (PDT)
Received: from htb-2n-eng-dhcp405.eng.vmware.com ([66.170.99.1])
        by smtp.gmail.com with ESMTPSA id
 o15sm48642199wrj.59.2019.04.15.15.14.24
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Mon, 15 Apr 2019 15:14:25 -0700 (PDT)
From: nadav.amit@gmail.com
To: Paolo Bonzini <pbonzini@redhat.com>
Cc: kvm@vger.kernel.org, Nadav Amit <nadav.amit@gmail.com>,
        Sean Christopherson <sean.j.christopherson@intel.com>
Subject: [kvm-unit-tests PATCH v2] x86: Remove INIT deassert
Date: Mon, 15 Apr 2019 15:14:21 -0700
Message-Id: <20190415221421.20903-1-nadav.amit@gmail.com>
X-Mailer: git-send-email 2.19.1
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

From: Nadav Amit <nadav.amit@gmail.com>

Currently, the INIT deassert is marked as an edge event instead of a
level, which is wrong. The deassert is actually not needed, as the
arbitration ID should have already been set after boot, and is not
affected by further INIT IPIs.

Remove the INIT deassert for these reasons.

Cc: Sean Christopherson <sean.j.christopherson@intel.com>
Signed-off-by: Nadav Amit <nadav.amit@gmail.com>
Reviewed-by: Sean Christopherson <sean.j.christopherson@intel.com>
---
 x86/cstart.S   | 1 -
 x86/cstart64.S | 1 -
 2 files changed, 2 deletions(-)

diff --git a/x86/cstart.S b/x86/cstart.S
index 143317e..79c5024 100644
--- a/x86/cstart.S
+++ b/x86/cstart.S
@@ -186,7 +186,6 @@ smp_init:
 	rep/movsb
 	mov $APIC_DEFAULT_PHYS_BASE, %eax
 	movl $(APIC_DEST_ALLBUT | APIC_DEST_PHYSICAL | APIC_DM_INIT | APIC_INT_ASSERT), APIC_ICR(%eax)
-	movl $(APIC_DEST_ALLBUT | APIC_DEST_PHYSICAL | APIC_DM_INIT), APIC_ICR(%eax)
 	movl $(APIC_DEST_ALLBUT | APIC_DEST_PHYSICAL | APIC_DM_STARTUP), APIC_ICR(%eax)
 	call fwcfg_get_nb_cpus
 1:	pause
diff --git a/x86/cstart64.S b/x86/cstart64.S
index 21db10f..e38767b 100644
--- a/x86/cstart64.S
+++ b/x86/cstart64.S
@@ -285,7 +285,6 @@ smp_init:
 	rep/movsb
 	mov $APIC_DEFAULT_PHYS_BASE, %eax
 	movl $(APIC_DEST_ALLBUT | APIC_DEST_PHYSICAL | APIC_DM_INIT | APIC_INT_ASSERT), APIC_ICR(%rax)
-	movl $(APIC_DEST_ALLBUT | APIC_DEST_PHYSICAL | APIC_DM_INIT), APIC_ICR(%rax)
 	movl $(APIC_DEST_ALLBUT | APIC_DEST_PHYSICAL | APIC_DM_STARTUP), APIC_ICR(%rax)
 	call fwcfg_get_nb_cpus
 1:	pause
